### PR TITLE
[BUG] Fixed JSON Format for Profile Creation and Modification

### PR DESCRIPTION
**Pull request by samrmur**
@RocketChat/android

Closes: #70

**Update Information**
- Added `SignUpPayload.kt`
- Changed `Login.kt` to account for changes

### CODE REVIEW

File: core/src/main/kotlin/chat/rocket/core/internal/model/SignUpPayload.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package chat.rocket.core.internal.model
+
+class SignUpPayload(val email: String?, val name: String?, val password: String?, val username: String?)
```
Comment (aniketsingh03) [inline core/src/main/kotlin/chat/rocket/core/internal/model/SignUpPayload.kt]: Take care of code formatting. One argument per line is usually preferred :smiley:

File: core/src/main/kotlin/chat/rocket/core/internal/model/UpdatePayload.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package chat.rocket.core.internal.model
+
+class UpdatePayload(val userId: String?, val data: UpdatePayload.Data) {
```
Comment (aniketsingh03) [inline core/src/main/kotlin/chat/rocket/core/internal/model/UpdatePayload.kt]: Here too!

**Comment by rafaelks** [comment]
@luciofm Could you please review this one?

File: core/src/main/kotlin/chat/rocket/core/internal/model/ProfileUpdatePayload.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package chat.rocket.core.internal.model
+
+class ProfileUpdatePayload(
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/model/ProfileUpdatePayload.kt]: Use `@JsonSerializable` annotation here.

File: core/src/main/kotlin/chat/rocket/core/internal/model/SignUpPayload.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package chat.rocket.core.internal.model
+
+class SignUpPayload(
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/model/SignUpPayload.kt]: Use `@JsonSerializable` annotation here.

**Comment by aniketsingh03** [comment]
@filipedelimabrito hasn't this already been done in https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/pull/84 ?

**Comment by leonardoaramaki** [comment]
@aniketsingh03 This https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/pull/84 already solves this. The main point of this PR would be separate the signup and update payloads. The latter is one valid point but I don't know if this PR is needed. Anyway, @samrmur can you please solve the conflicts?

**Comment by samrmur** [comment]
@leonardoaramaki Will do. The main reason for separating the payloads was because of the differences in payloads when registering versus when updating. See [rest-api/users/register/](https://rocket.chat/docs/developer-guides/rest-api/users/register/)

**Pull request merged**